driver-core/src/main/com/mongodb/internal/connection/DefaultConnectionPool.java

```diff
@@ -363,6 +397,44 @@ private boolean expired(final long startTime, final long curTime, final long max
         return maxTime != 0 && curTime - startTime > maxTime;
     }
 
+    private void connectionPoolCreated(final ConnectionPoolListener connectionPoolListener, final ServerId serverId,
```

Comment:
Looking at this fresh, I think these three methods together deserve a comment, like, 

     // send both current and deprecated events in order to preserve backwards compatibility